src/codegen/expression.rs

```diff
@@ -1043,38 +1087,8 @@ fn require(
     Expression::Poison
 }
 
-fn revert(
-    args: &[Expression],
-    cfg: &mut ControlFlowGraph,
-    contract_no: usize,
-    func: Option<&Function>,
-    ns: &Namespace,
-    vartab: &mut Vartable,
-    opt: &Options,
-) -> Expression {
-    let expr = args
-        .get(0)
-        .map(|s| expression(s, cfg, contract_no, func, ns, vartab, opt));
-    cfg.add(vartab, Instr::AssertFailure { expr });
-    Expression::Poison
-}
-
-fn self_destruct(
-    args: &[Expression],
-    cfg: &mut ControlFlowGraph,
-    contract_no: usize,
-    func: Option<&Function>,
-    ns: &Namespace,
-    vartab: &mut Vartable,
-    opt: &Options,
-) -> Expression {
-    let recipient = expression(&args[0], cfg, contract_no, func, ns, vartab, opt);
-    cfg.add(vartab, Instr::SelfDestruct { recipient });
-    Expression::Poison
-}
-
```

Comment:
We used to have an issue where some contracts would overflow the stack when solang is built in debug mode. The problem with the debug compilation path is that each variable will use up more stack space even if the liveness does not overlap with other variables.
The result of this is that `codegen::expression::expression` had a huge stack frame and we got stack overflows with large expression depths. That's why all these small functions exists: `codegen:::expression::expression` must be kept as short as possible.